File: app/src/main/java/pl/lisu188/speechrecorder/BootReceiver.java
package pl.lisu188.speechrecorder;

import android.app.Notification;
import android.app.NotificationChannel;
import android.app.NotificationManager;
import android.app.PendingIntent;
import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;

public class BootReceiver extends BroadcastReceiver {
    private static final String CHANNEL_ID = "speech_recorder_resume";
    private static final int NOTIFICATION_ID = 42;

    @Override
    public void onReceive(Context context, Intent intent) {
        boolean enabled = context.getSharedPreferences("recorder", Context.MODE_PRIVATE).getBoolean("enabled", false);
        if (!enabled) {
            return;
        }

        NotificationManager manager = context.getSystemService(NotificationManager.class);
        NotificationChannel channel = new NotificationChannel(CHANNEL_ID, "Wznowienie nagrywania", NotificationManager.IMPORTANCE_DEFAULT);
        channel.setDescription("Pozwala wznowić nasłuchiwanie po restarcie telefonu");
        manager.createNotificationChannel(channel);

        Intent open = new Intent(context, MainActivity.class).setAction(MainActivity.ACTION_RESUME_AFTER_BOOT);
        PendingIntent pending = PendingIntent.getActivity(context, 3, open, PendingIntent.FLAG_UPDATE_CURRENT | PendingIntent.FLAG_IMMUTABLE);
        Notification notification = new Notification.Builder(context, CHANNEL_ID)
                .setSmallIcon(R.drawable.ic_mic_notification)
                .setContentTitle("Speech Recorder zatrzymany po restarcie")
                .setContentText("Dotknij, aby wznowić stałe nasłuchiwanie mikrofonu")
                .setAutoCancel(true)
                .setContentIntent(pending)
                .build();
        manager.notify(NOTIFICATION_ID, notification);
    }
}

File: app/src/main/kotlin/pl/lisu188/speechrecorder/BootReceiver.kt
package pl.lisu188.speechrecorder

import android.app.Notification
import android.app.NotificationChannel
import android.app.NotificationManager
import android.app.PendingIntent
import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent

class BootReceiver : BroadcastReceiver() {
    override fun onReceive(context: Context, intent: Intent?) {
        val enabled = context.getSharedPreferences("recorder", Context.MODE_PRIVATE)
            .getBoolean("enabled", false)
        if (!enabled) return

        val manager = context.getSystemService(NotificationManager::class.java)
        manager.createNotificationChannel(
            NotificationChannel(
                CHANNEL_ID,
                "Wznowienie nagrywania",
                NotificationManager.IMPORTANCE_DEFAULT,
            ).apply {
                description = "Pozwala wznowić nasłuchiwanie po restarcie telefonu"
            },
        )

        val openIntent = Intent(context, MainActivity::class.java)
            .setAction(MainActivity.ACTION_RESUME_AFTER_BOOT)
        val pendingIntent = PendingIntent.getActivity(
            context,
            3,
            openIntent,
            PendingIntent.FLAG_UPDATE_CURRENT or PendingIntent.FLAG_IMMUTABLE,
        )

        val notification = Notification.Builder(context, CHANNEL_ID)
            .setSmallIcon(R.drawable.ic_mic_notification)
            .setContentTitle("Dyktafon zatrzymany po restarcie")
            .setContentText("Dotknij, aby wznowić stałe nasłuchiwanie mikrofonu")
            .setAutoCancel(true)
            .setContentIntent(pendingIntent)
            .build()

        manager.notify(NOTIFICATION_ID, notification)
    }

    private companion object {
        const val CHANNEL_ID = "speech_recorder_resume"
        const val NOTIFICATION_ID = 42
    }
}
